### PR TITLE
Make OfflineAudiocontext `Send` and `Sync`

### PR DESCRIPTION
**Pull request by orottier**
Regarding the discussion at #170
This is the first step, with a bit of reasoning, a safety wrapper and a small API change we can make the OfflineAudiocontext `Send` and `Sync`

### CODE REVIEW

File: src/context/mod.rs [hunk 1]
```diff
@@ -120,6 +120,12 @@ mod tests {
         require_send_sync_static(registration);
     }
 
+    #[test]
+    fn test_offline_audio_context_send_sync() {
```
Comment (orottier) [inline src/context/mod.rs:124]: Here's a nice (compile time) regression test for the issue

File: examples/decode_multithreaded.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+use float_eq::assert_float_eq;
+use std::sync::Arc;
+use web_audio_api::context::{BaseAudioContext, OfflineAudioContext};
+
+pub fn main() {
```
Comment (orottier) [inline examples/decode_multithreaded.rs:5]: I will add a better example than this, using a threadpool. If you spawn a thread per audio file things may run out of hand

File: examples/benchmarks.rs [hunk 3]
```diff
@@ -97,43 +97,43 @@ fn main() {
     {
         let name = "Baseline (silence)";
 
-        let mut context = OfflineAudioContext::new(2, DURATION * sample_rate as usize, sample_rate);
+        let context = OfflineAudioContext::new(2, DURATION * sample_rate as usize, sample_rate);
```
Comment (b-ma) [inline examples/benchmarks.rs:100]: Really like we can drop the `mut`

File: src/context/offline.rs [hunk 4]
```diff
@@ -70,15 +94,15 @@ impl OfflineAudioContext {
...
+            renderer: SingleUseRenderThread::new(renderer),
         }
     }
 
     /// Given the current connections and scheduled changes, starts rendering audio.
     ///
     /// This function will block the current thread and returns the rendered `AudioBuffer`
     /// synchronously. An async version is currently not implemented.
-    pub fn start_rendering_sync(&mut self) -> AudioBuffer {
+    pub fn start_rendering_sync(self) -> AudioBuffer {
```
[the remark this answers is not in the document]
Comment (b-ma) [reply src/context/offline.rs:105]: ok seems quite logical indeed

**Pull request merged**